File: iosApp/iosApp/AppViewModel.swift
//
//  AppViewModel.swift
//  iosApp
//
//  Created by Kim Fransman on 2021-10-19.
//  Copyright © 2021 Fransman. All rights reserved.
//

import Foundation
import SwiftUI
import shared

@MainActor
class AppViewModel: ObservableObject {
    @Published
    var players: [PlayersQuery.AllPlayer] = []

    let repository = ApolloRepository()

    func getPlayers() async {
        do {
            players = try await repository.getPlayers()
        } catch {
            print(error)
        }
    }
}

File: androidApp/src/main/java/se/fransman/tgapp/AppViewModel.kt
package se.fransman.tgapp

import androidx.lifecycle.ViewModel
import se.fransman.tg.ApolloRepository
import se.fransman.tg.PlayersQuery

class AppViewModel(private val repository: ApolloRepository): ViewModel() {
  suspend fun getPlayers(): List<PlayersQuery.AllPlayer> {
    return repository.getPlayers()
  }
}